Todo review

rachetm opened the pull request.

rachetm committed:
Adding all files

rachetm committed:
Fixed clear on add

rachetm committed:
Added redux functionality

rachetm committed:
Update README.md

rachetm committed:
Added search option

rachetm committed:
Fixed textfield not clearing on adding todo

rachetm committed:
Changed add.js filename to Add.js

rachetm committed:
Changed Readme

rachetm committed:
Merged adding-redux branch

rachetm committed:
Merged branch add-search-capability

rachetm committed:
Moved components/Add/Add.js to components/Add.js

rachetm committed:
Added search icon

rachetm committed:
Added hook in Add.js

rachetm committed:
Added props validation and made changes to app layout

aishwary08 commented on src/index.js:
```diff
@@ -0,0 +1,15 @@
+
+ReactDOM.render(
+  <Provider store={store}>
+    <App title="Todo App" />
```
Why is this title prop needed?

rachetm replied:
It's not necessary but it can be used to change the title from here.

aishwary08 commented on src/components/Search.js:
```diff
@@ -0,0 +1,33 @@
+    return (
+      <div id="searchField">
+        <i className="fas fa-search"></i>
+        <TextField
```
Better to use InputAdornment of TextField to add the left/right icons.

rachetm replied:
Will use that

aishwary08 commented on src/components/Add.js:
```diff
@@ -0,0 +1,65 @@
+}
+
+const mapStateToProps = state => {
+  return {
```
Are you intentionally trying to create a new ref of state? Can we directly return the state value?

rachetm replied:
It's not required at all. Removing it.

rachetm replied:
It's unnecessary. Removing it.

aishwary08 commented on src/components/Add.js:
```diff
@@ -0,0 +1,65 @@
+  }
+
+  return (
+    <div>
```
If you are wrapping children only for the purpose of returning one child, then you can avoid using extra div by using React.Fragment or <> (shorthand for the same) to wrap the children in one parent.

rachetm replied:
Removing the top level `<div>.` It's not required.

aishwary08 commented on src/components/Items/items.js:
```diff
@@ -0,0 +1,38 @@
+    const {items=[], filtered_items = []} = props;
+    const itemsToRender = filtered_items.length ? filtered_items : items;
+    
+    itemsToRender.forEach(item => {
```
Modify this logic. Conditionally set the check prop. Do not create another array and then append it later.

rachetm replied:
Cannot be done in any other way as discussed.

aishwary08 commented on src/components/Items/items.js:
```diff
@@ -0,0 +1,38 @@
+            );
+        else
+                unchecked_items.push(
+                    <Item key={item.id} {...item} onDel={props.onDel} check={props.check}/>
```
Why are you using props.check instead of item.checked?

rachetm replied:
That's a function and not the checked property in items object

aishwary08 commented on src/reducers/todos.js:
```diff
@@ -0,0 +1,59 @@
+const todos = (
+  state = {
```
Its better to give default values in each switch case. Otherwise as your app will grow you will have loads of default values. Use Destructuring and default values, you will be able to do it.

rachetm replied:
Okay. Will change that

aishwary08 commented on src/reducers/todos.js:
```diff
@@ -0,0 +1,59 @@
+            checked: false
+          }
+        ],
+        value: ""
```
Remove it if not used.

rachetm replied:
Done

aishwary08 commented on src/components/App.js:
```diff
@@ -0,0 +1,44 @@
+class App extends React.Component {
+  render() {
+    return (
+      <div style={{ height: "100%" }}>
```
Use Material's Grid instead of using flex property. You will reduce a lot of css.

rachetm replied:
Got it

rachetm replied:
Okay

aishwary08 commented on src/components/Add.js:
```diff
@@ -0,0 +1,65 @@
+          value={value}
+          autoFocus={true}
+        />
+        <Button variant="contained" id="addBtn" onClick={() => onClick()}>
```
You are creating extra function, that is not required. You can use onClick={onClick}

rachetm replied:
Alright

rachetm replied:
Got it

aishwary08 commented on src/components/Counter.js:
```diff
@@ -0,0 +1,38 @@
+function Counter(props) {
+  return (
+    <div className="counterContainer">
+      <p>
```
Use material's Typography component for displaying text and you can customise it easily.

rachetm replied:
Okay

aishwary08 commented on src/components/Items/item/item.js:
```diff
@@ -0,0 +1,38 @@
+          checked={this.props.checked}
+          onClick={() => this.props.check(this.props.id)}
+        />
+        <p className={this.props.checked? "checked" : ""}>{this.props.content}</p>
```
Use Typography component.

aishwary08 commented on src/components/Search.js:
```diff
@@ -0,0 +1,33 @@
+function Search(props){
+    return (
+      <div id="searchField">
+        <i className="fas fa-search"></i>
```
Component/element with no children can use self-closing bracket.

rachetm replied:
Got it

aishwary08 reviewed (changes requested):
1. Use destructuring and default values.
2. Avoid  using inline styles and id selector styles
3. Best to use class styles (Its easier for external consumer to override them)
4. Add comments in you code base
5. Can you come-up with a centralized implementation of mSTP and mDTP? Because it looks redundant.
6. Create separate CSS file for each component.
7. Prefer using const over let.
8. Take care of the code indentation
9. Avoid margins if you can, Prefer using Padding.
10. Remove Commented code before pushing for review. Also, its a good practice to review the code by yourself before pushing for final review. One of the way to do it is, by comparing your branch with master/dev (whichever is your main branch).

rachetm committed:
Made Requested changes

rachetm committed:
Minor fix

rachetm committed:
Removed unnecessary value key in state

rachetm committed:
Used input adornment for search icon

rachetm committed:
1. Removed unnecessary <div>
2. Changed onClick prop
3. Removed mapStateToProps function as it was not required

rachetm committed:
Modified package.json

rachetm committed:
Removed unnecessary styles used after using material-ui components

rachetm committed:
Changed title prop value

rachetm committed:
Moved default values to inside cases

rachetm committed:
Changed to Grid layout

rachetm committed:
Changed to grid layout and used IconButton component

rachetm committed:
Moved redundant code to a function

rachetm committed:
Used grid layout and Typography

rachetm committed:
Used grid layout and Typography
